### PR TITLE
CheckoutV2 tidying: Remove checkout v2 hook and settle sidebar styling

### PR DESCRIPTION
**Pull request by jjchrisdiehl**
This PR removes all of the `shouldHaveCheckoutV2`, `hasCheckoutVersion` and `useCheckoutV2` hooks that were used to A/B test v2 of our Checkout project - pbOQVh-45k-p2#comment-5821.

You'll notice the code is largely removal of the above hooks from different styles or conditional rendering blocks resulting in us falling back to the v1 Checkout experience.

Additionally, since we are collapsing v1 and v2 variants together, this PR handles a number of styling changes to ensure that v1 inherits any benefits from v2 - though we're largely falling back to v1 styles in this PR.

Related to https://github.com/Automattic/payments-shilling/issues/2754

## Testing Instructions

* We need to test a number of products here, including
    * Dotcom plans (including 100 Year plan)
    * Domains and Professional email
    * Akismet products
    * Jetpack products
    * DIFM
    * No selected site flow (such as http://calypso.localhost:3000/checkout/jetpack/jetpack_backup_t1_yearly)
    * Onboarding flows
* We ideally see the standard v1 experience along with the v1 sidebar. This includes a features list at the top, a list of products, cost overrides, the total, and any pertinent sidebar nudges (2 year upsells, DIFM next step list, renewal nudges to add soon to expire products to cart etc)

### CODE REVIEW

File: client/my-sites/checkout/src/components/checkout-sidebar-plan-upsell/index.tsx [hunk 1]
```diff
@@ -1,27 +1,17 @@
...
-import { ResponseCartProduct, useShoppingCart } from '@automattic/shopping-cart';
-import { hasCheckoutVersion } from '@automattic/wpcom-checkout';
-import styled from '@emotion/styled';
-import { Button } from '@wordpress/components';
+import { useShoppingCart } from '@automattic/shopping-cart';
 import { createElement, createInterpolateElement, useState } from '@wordpress/element';
 import { sprintf } from '@wordpress/i18n';
 import { useI18n } from '@wordpress/react-i18n';
 import debugFactory from 'debug';
-import { useTranslate } from 'i18n-calypso';
```
Comment (JessBoctor) [inline client/my-sites/checkout/src/components/checkout-sidebar-plan-upsell/index.tsx:17]: It seems like you are removing this translation function in favor of the `__()` translation string. Does this need to be imported?

It seems to work this way, so it isn't a blocker. I am more curious about why one method is preferred over the other. 

Screenshot 2024-06-18 at 15 54 36

File: client/my-sites/checkout/src/components/item-variation-picker/styles.tsx [hunk 2]
```diff
@@ -140,25 +132,18 @@ export const Variant = styled.div< { shouldUseCheckoutV2: boolean } >`
...
-export const Label = styled.span< { shouldUseCheckoutV2: boolean } >`
+export const Label = styled.span`
 	display: flex;
 	white-space: nowrap;
-
-	${ ( props ) =>
-		props.shouldUseCheckoutV2
-			? 'font-size: 14px; white-space: normal; text-align: start'
-			: 'font-size: inherit' };
+	font-size: inherit;
```
Comment (JessBoctor) [inline client/my-sites/checkout/src/components/item-variation-picker/styles.tsx]: I know on the call, one of the things you mentioned was making sure the font-size was consistent. Wouldn't declaring the font-size to be 14px, like it is done in other files, be the more consistent option than using `inherit` here?

Comment (JessBoctor) [reply client/my-sites/checkout/src/components/item-variation-picker/styles.tsx]: Currently, the labels are showing as 14px, but by being explicit here, you would prevent someone else from changing that.

File: client/my-sites/checkout/src/components/item-variation-picker/styles.tsx [hunk 3]
```diff
@@ -179,9 +164,7 @@ export const IntroPricingText = styled.span`
 	margin-bottom: 0rem;
 `;
 
-export const PriceTextContainer = styled.span< { shouldUseCheckoutV2: boolean } >`
-	${ ( props ) =>
-		props.shouldUseCheckoutV2
-			? 'font-size: 12px; text-align: initial;'
-			: 'font-size: inherit;	text-align: right;' };
+export const PriceTextContainer = styled.span`
+	font-size: inherit;
```
Comment (JessBoctor) [inline client/my-sites/checkout/src/components/item-variation-picker/styles.tsx]: Same question about the font sizes here.

File: packages/wpcom-checkout/src/field.tsx [hunk 4]
```diff
@@ -114,8 +110,7 @@ const Input = styled.input< {
 		${ ( props ) => ( props.isError ? props.theme.colors.error : props.theme.colors.borderColor ) };
 	padding: 7px ${ ( props ) => ( props.icon ? '60px' : '10px' ) } 7px 10px;
 	line-height: 1.5;
-
-	${ ( props ) => ( props.shouldUseCheckoutV2 ? `font-size: 14px;` : `font-size: 16px;` ) }
+	font-size: 16px;
```
Comment (JessBoctor) [inline packages/wpcom-checkout/src/field.tsx]: Looks like the form fields have a larger font size. This is fine if it is intentional, just thought I would point it out. 

Also, something to note, the form fields for credit card information and contact info (like if you are registering a domain) are in different measurements. The contact form uses rem, the credit card uses px, like above.

**Review by JessBoctor — APPROVED** [review]
Looks good. I have a few comments, but nothing that seems like it would really be a blocker. 

The only differences I could spot between checkout flows were how the refund information was presented under the submit button and how the multi-year upsells were styled. Otherwise, everything looked good and there weren't any untranslated strings

### Refund guarantee: 

**Jetpack Checkouts**
Screenshot 2024-06-18 at 16 41 58

**Other Checkouts**
Screenshot 2024-06-18 at 17 01 12

It was missing entirely from the DIFM checkout.
screencapture-wordpress-checkout-testingcheckout6-wordpress-com-2024-06-18-16_47_15

### Multi-year Upsell

**Jetpack & Akismet**
Screenshot 2024-06-18 at 16 43 08

**WordPress.com**
Screenshot 2024-06-18 at 17 00 12

Great work @jjchrisdiehl !

File: client/my-sites/checkout/src/components/item-variation-picker/styles.tsx (hunk 2 above)
Comment (jjchrisdiehl) [reply client/my-sites/checkout/src/components/item-variation-picker/styles.tsx]: That is a good idea. Ideally, we would pass this as a theme variable...but I haven't look further into that yet. May be a good follow up issue down the road.

File: packages/wpcom-checkout/src/field.tsx (hunk 4 above)
Comment (jjchrisdiehl) [reply packages/wpcom-checkout/src/field.tsx]: Yeah, this is where I wish I had more historical context about `/wpcom-checkout/` vs `/composite-checkout` vs `/my-sites/checkout` ... most likely they are all vestiges of past checkout redesigns that all intertwine now with mixed results. I think 14px here is correct, I'll take a look and update it where possible.

**Comment by jjchrisdiehl** [comment]
> The only differences I could spot between checkout flows were how the refund information was presented under the submit button and how the multi-year upsells were styled. Otherwise, everything looked good and there weren't any untranslated strings

This currently exists in production - Jetpack added their own submit footer (where the guarantee lives) and we added a more generic one. It's a good example of where different services could just feed this back into 'core' checkout and all services could benefit (i.e. default to off and then allow you to pass props for your service along with terms/content).

Might be worth a refactor in the future?

> It was missing entirely from the DIFM checkout. 

The two year upsell only currently applies to plans...sort of. DIFM has its own sidebar component which we don't show the upsell nudge on (even though they are technically buying a plan too).

Granted now that I look back on it this sidebar nudge code could use a refactor too!

> ### Multi-year Upsell

Again, Jetpack and Dotcom have diverged a bit here and have similar yet different versions of this component 🤷

File: client/my-sites/checkout/src/components/checkout-sidebar-plan-upsell/index.tsx (hunk 1 above)
Comment (jjchrisdiehl) [reply client/my-sites/checkout/src/components/checkout-sidebar-plan-upsell/index.tsx:17]: I'm not sure if I introduced the `translate` function to this file or if it was just a left over import that I was removing. I also do not know if there are any ad/disadvantages to `__` vs `translate` but I'll look into that and see! I think for our purposes both options work well enough.

Edit: Looks like the `translate` function is an older method we use (and still currently use) but we may change to the Gutenberg @i18n functions PCYsg-o1A-p2

File: packages/wpcom-checkout/src/field.tsx (hunk 4 above)
Comment (jjchrisdiehl) [reply packages/wpcom-checkout/src/field.tsx]: Also, the Payment section's fields are controlled by an iframe so I'll need to work out passing along styling information to the api.

**Pull request merged**